src/Listener/JsonApi/DocumentValidator.php

```diff
@@ -290,14 +290,11 @@ protected function _primaryDataMayHaveRelationships(): bool
             // multiple hasMany relationships
             $hasManys = $this->_getProperty($relationshipPathObject->dotted . '.data');
 
-            $i = 0;
```

Comment:
Thanks for your PR!

The previous file's change I agree with. This one not. This isn't an improvement. `foreach` is slower than `for` loop and it's really not any more difficult to read. This is purely a stylistic preference on your part.

Reply:
If `foreach` is slower, why would we stick with the slower implementation? My implementation uses `for` which you say is faster.

If you still want me to revert the changes in this file I will do that, but I feel what I have here is better.

Reply:
No, you're 100% right, I was looking at the PR in reverse. Too little sleep last night LOL

Reply:
Haha. No worries, I get it.